src/application-insights/azext_applicationinsights/util.py

```diff
@@ -0,0 +1,15 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from msrestazure.tools import is_valid_resource_id, parse_resource_id
+from azext_applicationinsights._client_factory import cf_components
+
+
+def get_id_from_azure_resource(cli_ctx, app):
+    if is_valid_resource_id(app):
+        parsed = parse_resource_id(app)
+        resource_group, name = parsed["resource_group"], parsed["name"]
+        return cf_components(cli_ctx, None).get(resource_group, name).app_id
```

Comment:
Would customers also want to run CRUD commands on a application?

Reply:
Yes, there's an open issue for this on azure-cli. It's out of scope for this PR however, i'm only covering data plane right now. My team *just* took ownership of the corresponding management plane as part of internal reshuffling. We are committed to fleshing out our support for management plane too, but one step at a time. 

I'll fix remaining comments.